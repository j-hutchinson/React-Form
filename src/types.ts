export enum FormState {
    USER = 'User',
    PRIVACY = 'Privacy',
    DONE = 'Done',
}

export const formItems = [FormState.USER, FormState.PRIVACY, FormState.DONE];

export interface ActionType {
    type: string;
    page?: FormState;
    privacy?: PrivacySettings;
    user?: UserSettings;
}

export interface ReduxState {
    page: FormState
    user: UserSettings;
    privacy: PrivacySettings;
}

export interface UserSettings {
    name: string;
    role: string;
    email: string;
    password: string
}

export interface PrivacySettings {
    updates: boolean;
    communication: boolean;
}
