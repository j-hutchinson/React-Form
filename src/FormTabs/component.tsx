import React from 'react';
import styled from 'styled-components'
import { formTabsList, FormState } from '../types';

interface Props {
    activeStep: FormState;
}

const StyledContainer = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
`;

const StyledTab = styled.div<{ active: boolean }>`
    background:  ${props => (props.active ? '#42daf5' : 'white')};
    border: 1px solid black;
    padding: 10px 0;
    text-align: center;
    font-size: 20px;
`;

const FormTabs = ({ activeStep }: Props): JSX.Element => (
    <StyledContainer>
        {formTabsList.map(tab =>
            <StyledTab active={activeStep === tab} key={tab}>
                {tab}
            </StyledTab>
        )}
    </StyledContainer>
);

export default FormTabs;