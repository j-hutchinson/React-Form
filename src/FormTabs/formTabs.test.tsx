import { shallow } from 'enzyme';
import React from 'react';
import FormTabs, { StyledTab } from './component';
import { FormState } from '../types';

describe('FormTabs component', () => {
    test('component matches snapshot', () => {
        expect.assertions(4);

        const wrapper = shallow(<FormTabs activePage={FormState.USER} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(StyledTab).at(0).props().active).toBe(true)
        expect(wrapper.find(StyledTab).at(1).props().active).toBe(false)
        expect(wrapper.find(StyledTab).at(2).props().active).toBe(false)
    });
});
