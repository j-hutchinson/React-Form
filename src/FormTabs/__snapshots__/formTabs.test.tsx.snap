// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTabs component component matches snapshot 1`] = `
<styled.div>
  <styled.div
    active={true}
    key="User"
  >
    User
  </styled.div>
  <styled.div
    active={false}
    key="Privacy"
  >
    Privacy
  </styled.div>
  <styled.div
    active={false}
    key="Done"
  >
    Done
  </styled.div>
</styled.div>
`;
